src/modules/common/containers/LanguageSelector.js

```diff
@@ -50,30 +52,32 @@ export class LanguageSelector extends React.Component {
             }
           }
         }
-        return goToCategories(city, languageCode)
+        return getCategoryPath(city, languageCode)
       case EVENTS_ROUTE:
         if (events && eventId) {
           const event = events.find(_event => _event.id === eventId)
           if (event) {
-            return goToEvents(city, languageCode, event.availableLanguages[languageCode])
+            return getEventPath(city, languageCode, event.availableLanguages[languageCode])
           }
         }
-        return goToEvents(city, languageCode)
+        return getEventPath(city, languageCode)
       case EXTRAS_ROUTE:
-        return goToExtras(city, languageCode, extraAlias)
+        return getExtraPath(city, languageCode, extraAlias)
       case DISCLAIMER_ROUTE:
-        return goToDisclaimer(city, languageCode)
+        return getDisclaimerPath(city, languageCode)
       case SEARCH_ROUTE:
-        return goToSearch(city, languageCode)
+        return getSearchPath(city, languageCode)
     }
   }
 
-  getSelectorItemModels () {
-    const languages = this.props.languages
+  getSelectorItemModels (): Array<SelectorItemModel> {
+    const {categories, events, location, languages} = this.props
     return languages && languages
       .map(language =>
         new SelectorItemModel({
-          code: language.code, name: language.name, href: this.getLanguageChangeAction(language.code)
+          code: language.code,
+          name: language.name,
+          href: LanguageSelector.getLanguageChangePath({categories, events, location, languageCode: language.code})
         })
       )
       .filter(selectorItem => selectorItem.href)
```

Comment:
I think we shouldn't filter these. I would expect to always see all languages, that the city offers, in the languageSelector. Once a city hasn't translated one category, then it wouldn't show up in the selector...
I'd prefer to go to a fallback url, like the categoriesHome of that language then (or at least disable it in the selector - like we do in the header when there are no events available)...
Maybe the latter suggestion would be more appropriate (and easier to implement :) )

Reply:
Alright, I don't like the first suggestion at all, but the second one sounds reasonable and makes sense for me, will do it like that :+1:

Reply:
On the other hand, it does not make sense to add links to sites in the header that don't exist, so filtering makes total sense for the hreflang links. The LanguageSelector is a different thing though.

Reply:
@Schedulaar done